lib/src/winrt/internal/async_helpers.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import '../../exceptions.dart';
+import '../foundation/enums.g.dart';
+import '../foundation/iasyncaction.dart';
+import '../foundation/iasyncinfo.dart';
+import '../foundation/iasyncoperation.dart';
+
+/// Completes the given [completer] when [asyncAction] completes.
+void completeAsyncAction(
+  IAsyncAction asyncAction,
+  Completer<void> completer,
+) =>
+    _completeAsyncDelegate(asyncAction, completer, completer.complete);
+
+/// Completes the given [completer] when [asyncOperation] completes.
+///
+/// If [asyncOperation] completes successfully, [completer] will be completed
+/// with the [value].
+void completeAsyncOperation<T, C>(
+  IAsyncOperation<T> asyncOperation,
+  Completer<C> completer,
+  C Function() value,
+) =>
+    _completeAsyncDelegate(
+        asyncOperation, completer, () => completer.complete(value()));
+
+/// Completes the given [completer] by polling [asyncDelegate]'s `status`
+/// property every `10` milliseconds until the [asyncDelegate] completes.
+///
+/// [onCompleted] is called when the [asyncDelegate] completes successfully.
+Future<void> _completeAsyncDelegate<T extends IAsyncInfo, C>(T asyncDelegate,
+    Completer<C> completer, void Function() onCompleted) async {
+  try {
+    while (asyncDelegate.status == AsyncStatus.started) {
+      // Yield execution to Dart for 10 milliseconds to avoid blocking the app.
+      await Future<void>.delayed(const Duration(milliseconds: 10));
```

Comment:
As an interim measure until there's a better solution, I think I'd suggest 1/60s as the poll frequency, to match the default frame rate of 60fps. More regular than that, and it's unlikely that the user will see a result for UI-centric content, and from a battery perspective, we know that the processor will be active at that frequency. 

(If we knew this was being called from Flutter, we might use `addPersistentFrameCallback`, but of course we don't have that guarantee.)

Reply:
Great, thanks!